refactor(oxlint): adjust ignore patterns by counting bytes instead of chars

Sysix opened the pull request:
`split_at` accepts the offset by bytes. 

Could not create a test where the chars vs bytes could matter:
https://github.com/oxc-project/oxc/blob/29516454d52d2197f9332b39162d27c913f31af6/apps/oxlint/src/lint.rs#L1026-L1039

camc314 commented:
### Merge activity

* **Jul 10, 10:17 PM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.
* **Jul 10, 10:17 PM UTC**: `camc314` added this pull request to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc).

Sysix committed:
refactor(oxlint): adjust ignore patterns by counting bytes instead of chars (#12209)

`split_at` accepts the offset by bytes.

Could not create a test where the chars vs bytes could matter:
https://github.com/oxc-project/oxc/blob/29516454d52d2197f9332b39162d27c913f31af6/apps/oxlint/src/lint.rs#L1026-L1039
